src/programs/vector/vector_translate.rs

```diff
@@ -0,0 +1,150 @@
+use std::borrow::Borrow;
+use std::ffi::CString;
+use std::ptr::{null, null_mut};
+
+use crate::{
+    errors::*,
+    utils::{_last_null_pointer_err},
+    Dataset,
+};
+use gdal_sys::{GDALVectorTranslate, GDALVectorTranslateOptions, GDALVectorTranslateOptionsFree};
+use libc::c_char;
+use crate::programs::destination::DatasetDestination;
+
+/// Wraps a [GDALVectorTranslateOptions] object.
+///
+/// [GDALVectorTranslateOptions]: https://gdal.org/api/gdal_utils.html#_CPPv426GDALVectorTranslateOptions
+pub struct VectorTranslateOptions{
+    c_options: *mut GDALVectorTranslateOptions
+}
+
+impl VectorTranslateOptions{
+    /// See [GDALVectorTranslateOptionsNew].
+    ///
+    /// [GDALVectorTranslateOptionsNew]: https://gdal.org/api/gdal_utils.html#_CPPv429GDALVectorTranslateOptionsNewPPcP35GDALVectorTranslateOptionsForBinary
+    pub fn new<S:Into<Vec<u8>>,I:IntoIterator<Item=S>>(args:I)->Result<Self>{
```

Comment:
Please run `cargo fmt`.

Reply:
It‘s excellent advice to use `CslString `instead of `fn new<S:Into<Vec<u8>>,I:IntoIterator<Item=S>>(args:I)->Result<Self> `, and I will make some modifications to this section.